core/src/lib.rs

```diff
@@ -116,6 +117,16 @@ mod tests {
         let mut kv_body: HashMap<String, Value> = HashMap::new();
         let mut kv_subject: HashMap<String, Value> = HashMap::new();
 
+        let type_rs = serde_json::to_value([
+            "VerifiableCredential".to_string(),
+            "PermanentResidentCard".to_string(),
+        ]);
+        if type_rs.is_ok() {
+            kv_body
+                .entry("type".to_string())
+                .or_insert(type_rs.unwrap());
```

Comment:
if we do `.unwrap()` it should be called as early as possible in the flow.

Reply:
Implemented with different data type